src/Files.App/Helpers/ContextFlyoutItemHelper.cs

```diff
@@ -509,45 +509,35 @@ private static bool Check(ContextMenuFlyoutItemViewModel item, CurrentInstanceVi
 				}.Build(),
 				new ContextMenuFlyoutItemViewModel
 				{
-					Text = "Archive".GetLocalizedResource(),
+					Text = "Compress".GetLocalizedResource(),
 					ShowInSearchPage = true,
 					OpacityIcon = new OpacityIconModel()
 					{
 						OpacityIconStyle = "ColorIconZip",
 					},
 					Items = new List<ContextMenuFlyoutItemViewModel>
 					{
-						new ContextMenuFlyoutItemViewModelBuilder(commands.DecompressArchive)
-						{
-							IsVisible = ArchiveHelpers.CanDecompress(selectedItems)
-						}.Build(),
-						new ContextMenuFlyoutItemViewModelBuilder(commands.DecompressArchiveHere)
-						{
-							IsVisible = ArchiveHelpers.CanDecompress(selectedItems)
-						}.Build(),
-						new ContextMenuFlyoutItemViewModelBuilder(commands.DecompressArchiveToChildFolder)
-						{
-							IsVisible = ArchiveHelpers.CanDecompress(selectedItems)
-						}.Build(),
-						new ContextMenuFlyoutItemViewModel
-						{
-							ShowItem = canDecompress && canCompress,
-							ItemType = ContextMenuFlyoutItemType.Separator,
-						},
-						new ContextMenuFlyoutItemViewModelBuilder(commands.CompressIntoArchive)
-						{
-							IsVisible = ArchiveHelpers.CanCompress(selectedItems)
-						}.Build(),
-						new ContextMenuFlyoutItemViewModelBuilder(commands.CompressIntoZip)
-						{
-							IsVisible = ArchiveHelpers.CanCompress(selectedItems)
-						}.Build(),
-						new ContextMenuFlyoutItemViewModelBuilder(commands.CompressIntoSevenZip)
-						{
-							IsVisible = ArchiveHelpers.CanCompress(selectedItems)
-						}.Build(),
+						new ContextMenuFlyoutItemViewModelBuilder(commands.CompressIntoArchive).Build(),
+						new ContextMenuFlyoutItemViewModelBuilder(commands.CompressIntoZip).Build(),
+						new ContextMenuFlyoutItemViewModelBuilder(commands.CompressIntoSevenZip).Build(),
 					},
-					ShowItem = itemsSelected
+					ShowItem = ArchiveHelpers.CanCompress(selectedItems)
```

Comment:
```suggestion
					ShowItem = itemsSelected && ArchiveHelpers.CanCompress(selectedItems)
```
Without `itemsSelected`, the compress menu will be displayed when right-clicking empty space.

Reply:
Doesn't `ArchiveHelpers.CanCompress` check for that?

Reply:
Currently, yes. This may be because it is possible to create an empty archive without selecting a file.